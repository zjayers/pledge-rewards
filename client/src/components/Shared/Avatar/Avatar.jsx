import React, { useEffect, useState } from 'react';
import ReactHtmlParser from 'react-html-parser';
import Avatar from './Avatar.style';
import { getUserImage } from '../../../services/apiService';

export default () => {
  const [img, setImg] = useState('');

  useEffect(() => {
    getUserImage().then((response) => {
      setImg(response.data);
    });
  }, []);

  return <Avatar>{ReactHtmlParser(img)}</Avatar>;
};
