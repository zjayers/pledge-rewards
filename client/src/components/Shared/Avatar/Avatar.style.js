import styled from 'styled-components';

export default styled.div`
  height: 7.2rem;
  width: 7.2rem;
  position: absolute;
  cursor: pointer;
  top: -3.4rem;
  left: 0;
  right: 0;
  margin: 0 auto;

  & img {
    height: 7.2rem;
    width: 7.2rem;
    border-radius: 100%;
    background-size: cover;
    z-index: 1;
  }
`;
