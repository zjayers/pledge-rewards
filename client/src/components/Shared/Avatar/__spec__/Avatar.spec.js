import React from 'react';
import { mount } from 'enzyme';

import toJson from 'enzyme-to-json';
import Avatar from '../Avatar';

describe('Avatar Component', () => {
  let component;

  beforeEach(() => {
    component = mount(<Avatar />);
  });

  afterEach(() => {
    component.unmount();
  });

  it('should render without failure', () => {
    expect(component).toBeDefined();
  });

  it('should match the test snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
