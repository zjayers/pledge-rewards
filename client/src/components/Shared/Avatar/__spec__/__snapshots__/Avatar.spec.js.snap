// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Component should match the test snapshot 1`] = `
<_default>
  <Avatarstyle___default>
    <div
      className="Avatarstyle___default-s7ub6l-0 iayjJg"
    />
  </Avatarstyle___default>
</_default>
`;
